docs/core/docker/docker-basics-dotnet-core.md

```diff
@@ -0,0 +1,280 @@
+
+   * The `OutputType` tag specifies that we're building an executable, in other words a console application.
+   * The `TargetFramework` tag specifies what .NET implementation we're targeting. In an advanced scenario, you can specify multiple target frameworks and build to the specified frameworks in a single operation. In this tutorial, we build for .NET Core 2.0.
+
+   `Program.cs`:
+
+   The program starts by `using System`. This statement means, "Bring everything in the `System` namespace into scope for this file." The `System` namespace includes basic constructs such as `string`, or numeric types.
+
+   We then define a namespace called `Hello`. You can change namespace to anything you want. A class named `Program` is defined within that namespace, with a `Main` method that takes an array of strings as its argument. This array contains the list of arguments passed in when the compiled program is called. In our example, the program only writes "Hello World!" to the console.
+
+2. `$ dotnet restore`
+
+   In .NET Core 2.x, **dotnet new** runs the [`dotnet restore`](../tools/dotnet-restore.md) command. **Dotnet restore** restores the tree of dependencies with a [NuGet](https://www.nuget.org/)(.NET package manager) call. 
+   NuGet performs the following tasks:
+   * analyzes the *Hello.csproj* file 
+   * downloads the file dependencies (or grabs from your machine cache)
+   * writes the *obj/project.assets.json* file
+   
+   The *project.assets.json* file is a complete set of the NuGet dependencies graph, binding resolutions, and other app metadata. This required file is used by other tools, such as [`dotnet build`](../tools/dotnet-build.md) and [`dotnet run`](../tools/dotnet-run.md), to correctly process the source code.
+   
+3. `$ dotnet run`
+
+   [`dotnet run`](../tools/dotnet-run.md) calls [`dotnet build`](../tools/dotnet-build.md) to confirm a successful build, and then calls `dotnet <assembly.dll>` to run the application.
+   
+    ```console
+    $ dotnet run
+    
+    Hello World!
+    ```
+
+    For advanced scenarios,  see [.NET Core Application Deployment](../deploying/index.md) for details.
+
+## Dockerize the .NET Core application
+
+The Hello .NET Core console app successfully runs locally. Now let's take it a step further and build and run the app in Docker.
+
+### Your first Dockerfile
+
+Open your text editor and let's get started! We're still working from the Hello directory we built the app in.
+
+Add the following Docker instructions for either Linux or [Windows Containers](https://docs.microsoft.com/en-us/virtualization/windowscontainers/about/) to a new file. When finished, save it in the root of your Hello directory as **Dockerfile**, with no extension (You may need to set your file type to `All types (*.*)` or something similar).
+
+```Dockerfile
+FROM microsoft/dotnet:2.0-sdk
+WORKDIR /app
+
+# copy csproj and restore as distinct layers
```

Comment:
Is this still needed, now that `dotnet publish` executes `dotnet restore`?

Reply:
It's from the .NET Docker Sample Dockerfile.

Reply:
I think the subsequent text should be explaining why `dotnet restore` is done explicitly.  It is done to help improve the iterative development experience.  If you make a code tweak and rebuild, the layers up to `COPY . .` don't have to be rebuilt and docker will utilize the cached layers.    See the [Build Cache](https://docs.docker.com/engine/userguide/eng-image/dockerfile_best-practices/#build-cache) section of the Dockerfile best practices for more details.